libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -956,36 +948,42 @@ string ABIFunctions::abiEncodingFunctionStruct(
 				{
 					if (storageSlotOffset != previousSlotOffset)
 					{
-						memberTempl("preprocess", "slotValue := sload(add(value, " + toCompactHexWithPrefix(storageSlotOffset) + "))");
+						members.back()["preprocess"] = "slotValue := sload(add(value, " + toCompactHexWithPrefix(storageSlotOffset) + "))";
```

Comment:
Since `preprocess` is part of the iteration of the outer template, we add it like that.